### PR TITLE
convert incoming labels to attributes

### PR DESCRIPTION
**Pull request by codeboten**
**Description:**
Converting incoming data that contains `Labels` to internal data containing `Attributes`. See the discussion [here](https://github.com/open-telemetry/opentelemetry-collector/pull/3740#discussion_r679612315) for more details

**Link to tracking Issue:** Part of #3535 

**Testing:** Updated unit tests

Follow up to https://github.com/open-telemetry/opentelemetry-collector/pull/3770

### CODE REVIEW

**Comment by bogdandrutu** [comment]
```
--- FAIL: TestExport (0.00s)
    otlp_test.go:63: 
        	Error Trace:	otlp_test.go:63
        	Error:      	Not equal: 
        	            	expected: pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc000486060)}
        	            	actual  : pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc000116060)}
        	            	
        	            	Diff:
        	Test:       	TestExport
FAIL
coverage: 13.3
FAIL	go.opentelemetry.io/collector/receiver/otlpreceiver/internal/metrics	0.015s
```

**Comment by codeboten** [comment]
> ```
> --- FAIL: TestExport (0.00s)
>     otlp_test.go:63: 
>         	Error Trace:	otlp_test.go:63
>         	Error:      	Not equal: 
>         	            	expected: pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc000486060)}
>         	            	actual  : pdata.Metrics{orig:(*v1.ExportMetricsServiceRequest)(0xc000116060)}
>         	            	
>         	            	Diff:
>         	Test:       	TestExport
> FAIL
> coverage: 13.3
> FAIL	go.opentelemetry.io/collector/receiver/otlpreceiver/internal/metrics	0.015s
> ```

Rebased on top of https://github.com/open-telemetry/opentelemetry-collector/pull/3770 to fix the failing tests, will mark ready for review on that PR is merged.

File: translator/internaldata/metrics_to_oc.go [hunk 1]
```diff
@@ -413,13 +413,13 @@ func labelValuesToOC(labels pdata.StringMap, labelKeys *labelKeysAndType) []*ocm
...
 	// Visit all defined labels in the point and override defaults with actual values
-	labels.Range(func(k string, v string) bool {
+	labels.Range(func(k string, v pdata.AttributeValue) bool {
 		// Find the appropriate label value that we need to update
 		keyIndex := labelKeys.keyIndices[k]
 		labelValue := labelValues[keyIndex]
 
 		// Update label value
-		labelValue.Value = v
+		labelValue.Value = v.StringVal() //TODO: should be converted here
```
Comment (tigrannajaryan) [inline translator/internaldata/metrics_to_oc.go:422]: Link TODO to https://github.com/open-telemetry/opentelemetry-collector/issues/3815

**Review by tigrannajaryan — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by codeboten** [comment]
> Please rebase

Done.

File: model/internal/otlp_wrapper.go [hunk 2]
```diff
@@ -55,6 +57,14 @@ func MetricsCompatibilityChanges(req *otlpcollectormetrics.ExportMetricsServiceR
...
 				case *otlpmetrics.Metric_IntSum:
 					metric.Data = intSumToSum(m)
+				case *otlpmetrics.Metric_Sum:
+					numberDataPointsLabelsToAttributes(m.Sum.DataPoints)
+				case *otlpmetrics.Metric_Gauge:
+					numberDataPointsLabelsToAttributes(m.Gauge.DataPoints)
+				case *otlpmetrics.Metric_Summary:
+					summaryDataPointsLabelsToAttributes(m.Summary.DataPoints)
+				case *otlpmetrics.Metric_Histogram:
+					histogramDataPointsLabelsToAttributes(m.Histogram.DataPoints)
```
[the remark this answers is not in the document]
Comment (codeboten) [reply model/internal/otlp_wrapper.go:67]: Are you suggesting the step to convert from `IntSum` to `Sum` and from `Labels` to `Attributes` should be separate? The current implementation does both at the same time. See the test here:

https://github.com/open-telemetry/opentelemetry-collector/blob/eadd5571f177803e004719b7705fdaf8f9d88137/model/internal/otlp_wrappers_test.go#L440-L480

**Review by bogdandrutu — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Is this ready for merging? I see open comments.

File: model/internal/otlp_wrapper.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (codeboten) [reply model/internal/otlp_wrapper.go:67]: @bogdandrutu I see you already approved, if you're ok with the current implementation please resolve this comment and merge :)

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@codeboten unitetests are failing.

**Comment by codeboten** [comment]
> @codeboten unitetests are failing.

Fixed the failing tests, PTAL.

**Pull request merged**